### PR TITLE
Stacked notifications

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**
Remove alerts in the `HeaderContainer` in favor of stacked dismissable notifications at the bottom right.

**Note:** Although I did some styling, it still needs a proper styling.

screenshot from 2018-03-14 17-28-02

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[Move all notifications to pills at the bottom of the screen](https://trello.com/c/dyGkNW61/124-move-all-notifications-to-pills-at-the-bottom-of-the-screen)

**Comment by dev-matan-tsuberi** [comment]
This probably needs more work and refactoring to make it show all pending transactions as in the design.

**Comment by tibetsprague** [comment]
Aaron has actually done styling for this on the front-end branch, so some reconciliation is in order

**Comment by tibetsprague** [comment]
and yes showing pending transactions can be a next step

File: src/actions/notificationsActions.ts [hunk 1]
```diff
@@ -1,13 +1,20 @@
-import { ActionTypes, INotificationsState } from "reducers/notificationsReducer";
+import { DismissNotification, ShowNotification } from "reducers/notificationsReducer";
 
-export const showAlert = (alert: string) => (dispatch: any) => {
-  const payload: INotificationsState = {
-    alert,
+export const showAlert = (message: string) => (dispatch: any) => {
+  const show: ShowNotification = {
+    type: "Notification/Show",
+    message,
```
Comment (tibetsprague) [inline src/actions/notificationsActions.ts]: lets stick with redux best practice of having just a type and a payload object in the action, so the message and the timestamp would both be part of the payload

File: src/components/Notification/Notification.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as moment from "moment";
+import * as React from "react";
+import * as css from "./Notification.scss";
+
+interface Props {
```
Comment (tibetsprague) [inline src/components/Notification/Notification.tsx]: ive been putting I in front of all interfaces, e.g. IProps, so lets stick with that

File: src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        <img
+          src="/assets/images/Icon/Close.svg"
+          alt="close"
+          className={css.close}
+          onClick={this.handleClose}/>
+      </div>
+    );
+  }
+
+  public handleClose = (e: any) => {
```
Comment (tibetsprague) [inline src/components/Notification/Notification.tsx]: ive been putting all render at the bottom of components and all other functions above, so lets stick with that

File: src/layouts/AppContainer.tsx [hunk 4]
```diff
@@ -9,37 +9,44 @@ import { ConnectionStatus, IWeb3State } from "reducers/web3Reducer";
...
 });
 
 interface IDispatchProps {
   connectToArc: typeof arcActions.connectToArc;
+  closeNotification: (id: number) => any;
 }
 
 const mapDispatchToProps = {
   connectToArc: arcActions.connectToArc,
+  closeNotification: notificationsActions.dismissAlert,
```
Comment (tibetsprague) [inline src/layouts/AppContainer.tsx]: lets use the same name for the function here, so dismissAlert should be closeNotification

File: src/reducers/notificationsReducer.ts [hunk 5]
```diff
@@ -1,21 +1,69 @@
-export enum ActionTypes {
-  ALERT_SHOW = "ALERT_SHOW",
+export interface Notification {
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: interface start with I

File: src/reducers/notificationsReducer.ts [hunk 6]
```diff
@@ -1,21 +1,69 @@
...
+  timestamp: Date;
 }
 
-export interface INotificationsState {
-  alert: string;
+export type NotificationState = Notification[];
+
+export type NotificationAction = ShowNotification | DismissNotification;
+
+export interface ShowNotification {
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: IShowNotification

File: src/reducers/notificationsReducer.ts [hunk 7]
```diff
@@ -1,21 +1,69 @@
...
+  type: "Notification/Show";
+  message: string;
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: lets use payload. we can start defining payload structures, probably a good idea.  I haven't been doing that so far

File: src/reducers/notificationsReducer.ts [hunk 8]
```diff
@@ -1,21 +1,69 @@
...
+  timestamp: Date;
+}
+
+export interface DismissNotification {
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: IDismissNotification, and use payload

Comment (tibetsprague) [reply src/reducers/notificationsReducer.ts]: payload in this case could just be the number

File: src/reducers/notificationsReducer.ts [hunk 9]
```diff
@@ -1,21 +1,69 @@
...
+  type: "Notification/Dismiss";
+  id: number;
 }
 
-export const initialState: INotificationsState = {
-  alert: "",
-};
+const example: NotificationState = [
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: what is this for?

File: src/reducers/notificationsReducer.ts [hunk 10]
```diff
@@ -1,21 +1,69 @@
...
+  },
+  {
+    id: 2,
+    message: "Hello 3",
+    timestamp: new Date(),
+  },
+];
+
+export const notificationsReducer =
+  (state: NotificationState = example, action: NotificationAction) =>
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: why are we starting with the example notifications?

File: src/reducers/notificationsReducer.ts [hunk 11]
```diff
@@ -1,21 +1,69 @@
...
+    action.type === "Notification/Show" ?
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: you like this better than a switch statement? id like to be consistent in whatever we do

File: src/reducers/notificationsReducer.ts [hunk 12]
```diff
@@ -1,21 +1,69 @@
...
+      [
+        ...state,
```
Comment (tibetsprague) [inline src/reducers/notificationsReducer.ts]: also lets be consistent and use immutability-helper

File: src/reducers/notificationsReducer.ts (hunk 11 above)
Comment (dev-matan-tsuberi) [reply src/reducers/notificationsReducer.ts]: Yes :). But I agree, let's be consistent.

**Comment by dev-matan-tsuberi** [comment]
Updated: https://github.com/daostack/alchemy/pull/35/commits/ad99e1ae7addca4b3d8c02bc8b4c6d437feff28d

**Pull request merged**